testsuite/cluster-test/src/cluster_swarm/cluster_swarm_kube.rs

```diff
@@ -0,0 +1,297 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use anyhow::{bail, format_err, Result};
+use async_trait::async_trait;
+
+use futures::future::try_join_all;
+use futures::lock::Mutex;
+use kube::{
+    api::{Api, PostParams},
+    client::APIClient,
+    config,
+};
+use slog_scope::*;
+use util::retry;
+
+use crate::cluster_swarm::ClusterSwarm;
+use crate::instance::Instance;
+use kube::api::ListParams;
+use libra_config::config::AdmissionControlConfig;
+
+const DEFAULT_NAMESPACE: &str = "default";
+
+const ERROR_NOT_FOUND: u16 = 404;
+
+pub struct ClusterSwarmKube {
+    client: APIClient,
+    validator_to_node: Arc<Mutex<HashMap<u32, Instance>>>,
+    fullnode_to_node: Arc<Mutex<HashMap<(u32, u32), Instance>>>,
+}
+
+impl ClusterSwarmKube {
+    pub async fn new() -> Result<Self> {
+        let mut config = config::load_kube_config().await;
+        if config.is_err() {
+            config = config::incluster_config();
+        }
+        let config = config.map_err(|e| format_err!("Failed to load config: {:?}", e))?;
+        let client = APIClient::new(config);
+        let validator_to_node = Arc::new(Mutex::new(HashMap::new()));
+        let fullnode_to_node = Arc::new(Mutex::new(HashMap::new()));
+        Ok(Self {
+            client,
+            validator_to_node,
+            fullnode_to_node,
+        })
+    }
+
+    fn validator_spec(
+        &self,
+        index: u32,
+        num_validators: u32,
+        num_fullnodes: u32,
+        node_name: &str,
+        image_tag: &str,
+        delete_data: bool,
+    ) -> Vec<u8> {
+        let pod_yaml = format!(
+            include_str!("validator_spec_template.yaml"),
+            index = index,
+            num_validators = num_validators,
+            num_fullnodes = num_fullnodes,
+            image_tag = image_tag,
+            node_name = node_name,
+            cfg_overrides = "",
+            delete_data = delete_data,
+        );
+        let pod_spec: serde_yaml::Value = serde_yaml::from_str(&pod_yaml).unwrap();
```

Comment:
nit: Our general guideline is to use `.expect` instead of `unwrap` everywhere except unit tests and `lock()`

Reply:
Updated.